docs/contributing.rst

```diff
@@ -71,12 +71,43 @@ or functions.
 Testing
 ^^^^^^^
 
-Twine is tested against Python versions 3.6, 3.7, and 3.8. To run these tests
-locally, you will need these versions of Python installed on your machine.
+We use `pytest`_ for writing and running tests.
+
+To run the tests in your virtual environment, run:
+
+.. code-block:: console
+
+  tox -e py
+
+To pass options to ``pytest``, e.g. the name of a test, run:
+
+.. code-block:: console
+
+  tox -e py -- tests/test_upload.py::test_exception_for_http_status
+
+You can also set the ``PYTEST_ADDOPTS`` environment variable to use the same
+options on every test run. For example, to skip integration tests and turn
+off the coverage report, run:
+
+.. code-block:: console
+
+  export PYTEST_ADDOPTS=-k "not integration" --cov-report=
```

Comment:
I tried this command locally on my mac in a virtualenv, and it didn't work.

```
$ export PYTEST_ADDOPTS=-k "not integration" --cov-report=
-bash: export: `not integration': not a valid identifier
-bash: export: `--cov-report=': not a valid identifier
```

Reply:
Good catch, @deveshks. Fixed in https://github.com/pypa/twine/pull/633/commits/e9fce0f17f266f3e4417d68d8ae7d1933582659b.

I will address questions about the use of `PYTEST_ADDOPTS` in one of the other comment threads.

Reply:
> Good catch, @deveshks. Fixed in [e9fce0f](https://github.com/pypa/twine/commit/e9fce0f17f266f3e4417d68d8ae7d1933582659b).

Cool, the export works now.

> I will address questions about the use of `PYTEST_ADDOPTS` in one of the other comment threads.

Thanks, I am also interest in knowing how to use `PYTEST_ADDOPTS` with tox.